### PR TITLE
Add service for listing reject files

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        CloudBlobClient cloudBlobClient = account.createCloudBlobClient();
+
+        this.blobManager = new BlobManager(cloudBlobClient, blobManagementProperties);
+
+        this.rejectedContainer = cloudBlobClient.getContainerReference("test-rejected");
+        this.rejectedContainer.createIfNotExists();
+    }
+
+    @Test
+    public void should_delete_old_files() throws Exception {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java]: this method name needs changing ?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.List;
+
+import static com.microsoft.azure.storage.blob.BlobListingDetails.SNAPSHOTS;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RejectedEnvelopesReportServiceTest {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java]: can we please add a test case when there are no rejected envelopes ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java]: oops!

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java (hunk 2 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java]: done

**Pull request merged**